### PR TITLE
feat(ruleset-bundler): add ruleset loader

### LINKED ISSUE #1956: Expose Javascript function to load custom ruleset from file
**User story.**
As a **developer consuming spectral-core from javascript**, I can **load custom ruselet from yml files**, so that I can use them programatically on a microservice.

**Is your feature request related to a problem?**
I'm trying to use spectral-core 6 from javascript code and I'm not able to programatically load rulesets from yml files.
Loading files as yml files doesn't work as it doesn't resolve functions (like truthy, falsy...) and all I get are string properties...

Digging in spectral-core source code there is a function for reading files is included in spectral-cli but that function is not exported as a type.d.ts (or is it?).

https://github.com/stoplightio/spectral/blob/develop/packages/cli/src/services/linter/utils/getRuleset.ts#L36

Trying to import the function like this doesn't work:
```ts
import { getRuleset } from '@stoplight/spectral-cli/dist/services/linter/utils';
```
(I guess I can not import from _dist_ folder just like that)

**Describe the solution you'd like**

I would like spectral-core to expose `getRuleset` function for other applications to use it programatically
or even update `Spectral.setRuleset` signature so it can receive rules as string urls

```ts
Spectral.setRuleset(ruleset: RulesetDefinition | Ruleset | string): void
```

```ts
const fileUrl = '<local or http file url>';
const spectral = new Spectral();
spectral.setRuleset(fileUrl);
spectral.run(..);
```

**Additional context**

### PR DESCRIPTION
**Pull request by P0lip**
Closes #1956

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ commands:
       - run: yarn pretest
       - run:
           name: Run node tests
-          command: yarn test.jest --coverage --maxWorkers=<< parameters.max-workers >>
```
Comment (marbemac) [inline .circleci/config.yml]: This removal of `--coverage` intentional? (ok if it is, just double checking)

File: packages/ruleset-bundler/README.md [hunk 2]
```diff
@@ -13,3 +11,24 @@
 - treeshake - whether to enable tree shaking. False by default.
 
 **Bolded** options are required.
+
+## Loading YAML/JSON Ruleset
+
+It is assumed you have [@stoplight/spectral-ruleset-migrator](https://www.npmjs.com/package/@stoplight/spectral-ruleset-migrator) installed.
```
Comment (marbemac) [inline packages/ruleset-bundler/README.md]: Any reason to not have this as a dependency if it's always required? Env issue, dedupe issue, something else?

File: packages/ruleset-bundler/package.json [hunk 3]
```diff
@@ -18,6 +18,11 @@
     ".": {
       "default": "./dist/index.js"
     },
+    "./with-loader": {
+      "default": "./dist/loader/node.js",
+      "node": "./dist/loader/node.js",
+      "browser": "./dist/loader/browser.js"
```
Comment (marbemac) [inline packages/ruleset-bundler/package.json:24]: 👌

File: packages/ruleset-bundler/src/loader/common/bundle.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { ErrorWithCause } from 'pony-cause';
+
+import { stdin } from '../../plugins/stdin';
+import { isBasicRuleset, isErrorWithCode } from './utils';
+import { bundleRuleset, BundleOptions, IO } from '../../index';
+
+export async function bundle(rulesetFile: string, bundleOptions: BundleOptions, { fs }: IO): Promise<string> {
+  try {
+    if (isBasicRuleset(rulesetFile)) {
+      const migratedRuleset = await migrateRuleset(rulesetFile, {
```
Comment (marbemac) [inline packages/ruleset-bundler/src/loader/common/bundle.ts:12]: Ah I think I see - `migrateRuleset` naming confused me. More like a ruleset transformer in this case, to transform from plain object format to esm format.

**Review by marbemac — COMMENTED** [review]
excellent work, mostly just a couple of questions

File: .circleci/config.yml (hunk 1 above)
Comment (P0lip) [reply .circleci/config.yml]: Yeah, it's already gone in develop branch

File: packages/ruleset-bundler/README.md (hunk 2 above)
Comment (P0lip) [reply packages/ruleset-bundler/README.md]: > Any reason to not have this as a dependency if it's always required? Env issue, dedupe issue, something else?

hmmm, yeah, we could have it installed, why not.

**Pull request merged**